### DISCUSSION TITLE
Rationale for name change?

### DISCUSSION THREAD

**Discussion by uvtc** [libui-ng/libui-ng#49] [General]
Hi,

I'm curious about the rationale for a name change (from libui to libui-ng). Are there plans to diverge substantially from libui?

**Comment by cody271** [comment]
Hi @uvtc!

> I'm curious about the rationale for a name change (from libui to libui-ng).

The name change is both a practical matter and tradition:

a) As the original libui repository is abandoned, the community no longer had access to upstream. No way to move forward without a new repository. This time owned under an organization, instead of a personal account.

Also someone else appears to be squatting on @libui organization name, that's not any of us.
(And post acquisition, GitHub support policy is to no longer honor their original "no name squatting" policy.)

b) Appending [`-ng`](https://en.wikipedia.org/wiki/Ng#Other_uses) to the name of a community fork of a software project is a long standing free software tradition.
In this case it means "libui for the next generation". Some other examples of this: [syslog-ng](https://www.syslog-ng.com/) and [XCP-ng](https://xcp-ng.com/).

> Are there plans to diverge substantially from libui?

Short answer would be "no", but that depends on what you mean exactly?

We are continuing the project in largely the same direction: a portable GUI library for C.
You can track the progress on merging all the old PRs [here](https://github.com/libui-ng/libui-ng/issues/8). This is the first step.
Then the features we plan to add shortly thereafter, see all the issues tagged [`enhancement`](https://github.com/libui-ng/libui-ng/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement).

**Reply by uvtc** [reply]
> > Are there plans to diverge substantially from libui?
>
> Short answer would be "no", but that depends on what you mean exactly?

Ah. I wasn't familiar with the custom, and had thought that the "-ng" name change might imply that changes in direction were afoot.

I didn't know that the libui org name was already taken. Given that though, I had assumed that the project would keep its name: that is, using "github.com/libui-ng/libui". The "-ng" would be a misnomer if, going forward, libui-ng becomes the canonical one and folks wonder why it's called "next generation" since everyone's using it in that future "*current* generation". :) If one had to rename every time a project was forked, a very large number of projects would have that "-ng" suffix. Especially if the original forked-from project appears to be no longer maintained.

Thanks so much for your work on this new fork!

**Reply by cody271** [reply]
> folks wonder why it's called "next generation" since everyone's using it in that future "current generation". :) If one had to rename every time a project was forked, a very large number of projects would have that "-ng" suffix. Especially if the original forked-from project appears to be no longer maintained.

Well, that is in fact what has happened over the years: 
https://askubuntu.com/questions/46596/what-does-the-ng-suffix-mean-in-package-names

Best example is `syslog-ng`, which has been the main implementation of standalone "syslog" since.. 1998.

Their repository is _still_ named [`syslog-ng/syslog-ng`](https://github.com/syslog-ng/syslog-ng), decades later. 🤣
